docker-entrypoint.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/sh -e
+
+### Usage: /docker-entrypoint.sh <command> <args>
+command=$1
+
+case "$command" in
+  serve)
+    for file_candidate in $@ ; do
+      if test -f "$file_candidate"; then
```

Comment:
I wonder if it would be better to use e.g. `serve --template TEMPLATE ...`  instead of interpreting any existing file as a template.
This would make it more explicit, and could avoid potential esoteric bugs.

Reply:
The advantage of the implicit approach is avoiding unnecessary complications.

Thinking about this, adding new flags to the entrypoint has its pros and cons. Let's talk more about the disadvantages of using a flag:
* It's hard to document them.
* Users can't see them through help.
* Interference of the entrypoint is too big.
* The serve command has additional flags for addresses to listen on, which leads us to parse flags two times (with the entrypoint and with dex).

Maybe we can add a new *command* instead (something like "serve-with-template"), or use some env variable? The question is open.

Reply:
You're right with the disadvantages. 
Then I would suggest to either leave it as is, or to use `serve(-with|-from)-template`

Reply:
I'd really like to avoid adding templating to the Dex binary.

Reply:
An example of the solution described by heidemn can be found here https://github.com/dexidp/dex/pull/1965. We closed that PR in favor of #1893.